R/tbl_propdiff.R

```diff
@@ -0,0 +1,365 @@
+#' Calculating unadjusted and adjusted differences in rates
+#'
+#' This function calculates the unadjusted or adjusted difference in rates with confidence interval.
+#'
+#' @param data A data frame
+#' @param outcome A vector of outcome variable names. Outcome variables can be numeric, character or factor, but must have two and only two non-missing levels
+#' @param predictor The name of the predictor (stratifying) variable. This can be numeric, character or factor, but must have two and only two non-missing levels
+#' @param covariates A vector of variable names that the difference in rates will be adjusted for. The p-value displayed will be from a multivariable logistic regression model containing these covariates. The default option is `NULL` which calculates an unadjusted difference in rates.
+#' @param method The method for calculating confidence intervals around the adjusted difference in rates. The 'centile' option is the default and uses 2000 bootstrap resamples by default and calculates the confidence interval using centiles (by default, 2.5th and 97.5th centiles for a 95% confidence interval). The 'mean' option can also be selected, which uses a default 200 bootstrap resamples and uses the standard deviation for the mean adjusted difference across all resamples to calculate the confidence interval.
+#' @param conflevel Confidence level of the returned confidence interval. Must be a single number between 0 and 1. The default is a 95% confidence interval.
+#' @param rev By default, the rates and difference in rates are presented as group 1, group 2, (group 1 - group 2). To reverse the order of the columns and how the difference is calculated, set this option to `TRUE`
+#' @param bootstrapn The number of bootstrap resamples to use. The default is 2000 when `method = "centile"` and the default is 200 when `method = "mean"`
+#' @param estimate_fun Function to round and format estimates. By default `style_sigfig`, but can take any formatting function
+#' @param pvalue_fun Function to round and format p-value. By default `style_pvalue`, but can take any formatting function
+#'
+#' @return A formatted `gt` table with the unadjusted rate in each group, and the difference in rates, confidence interval and p-value for the comparison (unadjusted by default, adjusted if `covariates` option specified)
+#' @export
+#'
+#' @examples
+#' tbl_propdiff(
+#' data = trial,
+#' outcome = "response",
+#' predictor = "trt"
+#' )
+#'
+#' tbl_propdiff(
+#' data = trial,
+#' outcome = "response",
+#' predictor = "trt",
+#' covariates = c("age", "stage"),
+#' method = "mean",
+#' bootstrapn = 250
+#' )
+
+tbl_propdiff <- function(data, outcome, predictor, covariates = NULL,
+                         method = c("centile", "mean"),
+                         conflevel = 0.95, rev = FALSE,
+                         bootstrapn = ifelse(method == "centile", 2000, 200),
+                         estimate_fun = function(x) style_sigfig(x),
```

Comment:
this can just be `estimate_fun = style_sigfig`

Reply:
updated this